### PR TITLE
feat(apm): Improve APM profiling

### PR DESCRIPTION
**Pull request by HazAT**
Add React Profiler to LoadingIndicator
Also adds `data-entry="true"` to your app script so we can measure the script evaluation time.

### CODE REVIEW

**Review by rhcarvalho — DISMISSED** [review]

File: tests/js/spec/components/assigneeSelector.spec.jsx [hunk 1]
```diff
@@ -136,7 +136,7 @@ describe('AssigneeSelector', function() {
 
   it('should initially have loading state', function() {
     openMenu();
-    expect(assigneeSelector.find('LoadingIndicator')).toHaveLength(1);
+    expect(assigneeSelector.find('LoadingIndicator')).toHaveLength(2);
```
Comment (rhcarvalho) [inline tests/js/spec/components/assigneeSelector.spec.jsx]: Do we really expect to have 2x "LoadingIndicator"?
Is the profiler instance confused with the LoadingIndicator itself? If so we could try to fine tune the selector.

Comment (priscilawebdev) [inline tests/js/spec/components/assigneeSelector.spec.jsx]: are you sure to change this to `2` I believe we should not get 2 LoadingIndicator nodes.

**Comment by HazAT** [comment]
@priscilawebdev I removed the bump of the JS SDK to here https://github.com/getsentry/sentry/pull/17576
So we can merge this independently

**Comment by HazAT** [comment]
I am adding @billyvg here since I think you did the first implementation of the profiler.
Are those valid concerns or do we just need to change the tests?

**Comment by HazAT** [comment]
So @billyvg to summarize:

We are experiencing this:
image (4)

We get duplicated components, the page looks fine and I think the `Profiler` mirrors the name, but we should probably change this so the Profiler is actually called `<Profiler />`

**Comment by priscilawebdev** [comment]
@HazAT here are my findings:

Everything is fine with the HOC `profiler()`. I checked the DOM and we only render one component.

The problem is related to Jest. Please see: https://github.com/enzymejs/enzyme/issues/539

If I do for example:

```
   const wrapper = shallow(<LazyLoad component={getComponent} />);
   const loadingIndicatorElement = wrapper.find('LoadingIndicator').shallow()
 expect(loadingIndicatorElement).toHaveLength(1);
```

Test will pass! ✅

Now I'm wondering if there is a better way...

File: src/sentry/static/sentry/app/utils/profiler.jsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ export default function profiler() {
     const displayName = getDisplayName(WrappedComponent);
 
     return class extends React.Component {
-      static displayName = displayName;
+      static displayName = `profiler(${displayName})`;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/utils/profiler.jsx]: the problem was resolved after updating this line 😉

**Review by billyvg — APPROVED** [review]

**Pull request merged**